### ISSUE TITLE
Resize Not Working As Expected

### ISSUE THREAD

**Issue by MONO77** [koekeishiya/yabai#1207]
Hi guys, 

Here is a [short video demo](https://www.dropbox.com/s/0zc7y53sw2i5wwo/Yabai%20Resize%20Issue%20Demo.mp4?dl=0) of my problem.

Basically, for the first part when I execute `yabai -m window --resize right:-20:0` multiple times on the **middle window** via keybinding, instead of moving **right border** of the **middle window** to the **left**, the **right border** of the **left-most** window is moved, which quite confuses me.

Similarly, when I try to drag the **right border** of **middle window** to the **right** to resize, not only the **right border** of **middle window** goes to right, its **left border** also goes to **right** (i.e. shrinks). 

I am not sure if I have misunderstood the command, but as far as I have read the man page and wiki, the behavior of resize is not as expected.

Any help would be appreciated. Thanks in advance.

**Comment by davelosert** [comment]
I am currently facing this exact same issue.

## My observations

1. This only happens if the resizing is done on the border of two 'nodes'. e.g. in the video one large node is the `yabai config` (first child) and the `terminal` and `finder` windows, and another one is the `browser` (second child). 

    So in the video, the shrinking of the right side of the terminal is done between those two nodes.

    If one would resize to the left of the terminal window (or to the right of the yabai config window for that matter), it wouldn't happen as it is a resize within a node itself (at least, it is like this for me).

2. The sum of pixels that an entire 'node' grows or shrinks is still correct, so in the video the browser window becomes 20 pixels larger indeed - but within the node, it just seems to be randomly applying the resizing events to reach that sum (maybe like 100 pixels of the `yabai config` and another 100 for the terminal?). I didn't find the ratio or a pattern here (yet).


## My guess

This is because the resizing of windows doesn't actually happen on the window itself, but on the parent-node (if it exists) which I believe happens here: https://github.com/koekeishiya/yabai/blob/master/src/window_manager.c#L475

So basically the resizing happens on the parent node, which I guess fires events for the children (the windows) to resize accordingly, and somewhere then it becomes a bit random which window actually grows or shrinks.

I am too unfamiliar with C to really confirm this suspicion or really trace down the bug to the source - but maybe my observation helps in fixing this?
